### PR TITLE
chore(components): rollup to vite migration (VIV-1263)

### PR DESCRIPTION
**Pull request by YonatanKra**
chore(components): rollup to vite migration (VIV-1263)

### CODE REVIEW

File: docs/ui-test/readme.md [hunk 1]
```diff
@@ -56,12 +56,6 @@ After you have docker installed, run:
 
 `npm run nx e2e components`
```
Comment (rachelbt) [inline docs/ui-test/readme.md:58]: no more docker needed? :)

File: libs/components/src/lib/date-picker/date-picker.ts [hunk 2]
```diff
@@ -30,6 +30,16 @@ import { buildMonthPickerGrid, MonthsPerRow } from './calendar/monthPickerGrid';
  * @public
  */
 export class DatePicker extends FoundationElement {
+	#dismissOnClickOutside = (event: MouseEvent) => {
```
Comment (rachelbt) [inline libs/components/src/lib/date-picker/date-picker.ts:33]: what this has to do with this PR?

File: libs/components/tsconfig.json [hunk 3]
```diff
@@ -34,3 +34,4 @@
     }
   ]
 }
+
```
Comment (rachelbt) [inline libs/components/tsconfig.json:37]: really?
Dont we have enough files here as it is? :) :) :)

**Review by rachelbt — COMMENTED** [review]
- went over the files, can say I understood all the code...
- there's no deploy in the PR
- I tried running the branch locally but got this:
```
[!] Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/accordion-item/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/accordion-item/index.js'
Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/accordion-item/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/accordion-item/index.js'
✓ 1262 modules transformed.
rendering chunks...
bundles vivid-components → dist/apps/docs/assets/scripts...
[!] Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/fab/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/fab/index.js'
Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/fab/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/fab/index.js'
bundles vivid-components → dist/apps/docs/assets/scripts...
[!] Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js'
Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js'
bundles vivid-components → dist/apps/docs/assets/scripts...
[!] Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js'
Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js'
bundles vivid-components → dist/apps/docs/assets/scripts...
[!] Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js'
Error: Could not load /Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js (imported by virtual:vivid-components): ENOENT: no such file or directory, open '/Users/RTannenbaum/projects/vivid-3/dist/libs/components/listbox/index.js'
```
and yes - I run: `rm -rf node_modules && npm i`
tried to generate new PNG - all went well 👍

File: libs/components/src/lib/date-picker/date-picker.ts (hunk 2 above)
Comment (YonatanKra) [reply libs/components/src/lib/date-picker/date-picker.ts:33]: New ts version doesn't allow to use private methods before they are declared on the class (I think it's a bug in TS, but this is how it works).

**Comment by YonatanKra** [comment]
@rachelbt 
- Deploy - it's a caching issue. I'll look at it. Good point.
- The error you see is acceptable for `dev` only. The problem is a race condition between our build and the 11ty dev server. Can you see anything amiss in the local dev?

**Comment by rachelbt** [comment]
> * The error you see is acceptable for `dev` only. The problem is a race condition between our build and the 11ty dev server. Can you see anything amiss in the local dev?

the local env works OK.
The only thing is - after changing files - the browser is loading forever - you need to stop and refresh again to see the change (in css or HTML, not tests :) ). BTW - This was also before the change you made.

File: .github/workflows/_visual-regression.yml [hunk 4]
```diff
@@ -5,9 +5,10 @@ on: workflow_call
 jobs:
   test:
     runs-on: ubuntu-latest
-    container: vivid3/vonage:2.0.1
+    container:
+      image: drizzt99/vonage:2.1.0
```
Comment (rinaok) [inline .github/workflows/_visual-regression.yml:9]: what is drizzt99?

**Review by rinaok — APPROVED** [review]

File: .github/workflows/_visual-regression.yml [hunk 5]
```diff
@@ -5,9 +5,10 @@ on: workflow_call
...
     steps:
-      - run: echo "Running vivid3/vonage:2.0.1"
+      - run: echo "Running drizzt99/vonage:2.1.0"
```
[the remark this answers is not in the document]
Comment (rachelbt) [reply .github/workflows/_visual-regression.yml:11]: https://jira.vonage.com/browse/VIV-1276

**Pull request merged**